src/endpoints/common_endpoint_registry.cpp

```diff
@@ -226,30 +226,8 @@ namespace ccf
 
     auto is_tx_committed =
```

Comment:
This function has to be defined by all apps using historical endpoints, even though it is very much framework wide. Could the `available` argument of `adapter()` default to `ccf::historical::is_tx_committed` instead (just like for `extractor`)?

Reply:
I wondered that, but for it to work, the adapter would need access to the consensus somehow instead of being a pure function.

Reply:
That's right, this is a functor purely so it can capture `consensus`. Consensus could be a singleton, or otherwise globally-available, but we've avoided that so far.

Reply:
Ok, let's wrap up #2640 first and think about this separately. I would prefer to have the adapter as a frontend mixin and avoid singletons in the code base as much as possible.